src/coreclr/src/vm/prestub.cpp

```diff
@@ -2169,6 +2169,11 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT, CallerGCMode callerGCMo
             }
         }
 
+        // P/Invoke marked with UnmanagedCallersOnlyAttribute is not
+        // presently supported.
+        if (HasUnmanagedCallersOnlyAttribute())
```

Comment:
CreateNDirectStubWorker may be a better place for this

Reply:
(It is where other similar checks for invalid signatures, etc. happen.)

Reply:
I figured putting this in the PreStub was the best place to catch all uses. Am I handling more places by putting it lower in the stack?

Reply:
Putting this in PreStub is penalizing the crossgen case. For crossgen, we want to just lookup the code, without doing any error checking. If we find good pregenerated code, we know that the error checks succeeded during crossgen, no need to repeat them.